extensions/desktop-common/desktop-exports

```diff
@@ -0,0 +1,416 @@
+#!/bin/bash -e
+
+###############################################
+# Launcher common exports for any desktop app #
+###############################################
+
+function prepend_dir() {
+  local var="$1"
+  local dir="$2"
+  if [ -d "$dir" ]; then
+    eval "export $var=\"\$dir\${$var:+:\$$var}\""
+  fi
+}
+
+function append_dir() {
+  local var="$1"
+  local dir="$2"
+  if [ -d "$dir" ]; then
+    eval "export $var=\"\${$var:+\$$var:}\$dir\""
+  fi
+}
+
+function can_open_file() {
+  head -c0 "$1" &> /dev/null
+}
+
+function update_xdg_dirs_values() {
+  local save_initial_values=false
+  local XDG_DIRS="DOCUMENTS DESKTOP DOWNLOAD MUSIC PICTURES VIDEOS PUBLICSHARE TEMPLATES"
+  unset XDG_SPECIAL_DIRS_PATHS
+
+  if [ -f "${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs" ]; then
+    # shellcheck disable=SC1090
+    source "${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs"
+  fi
+
+  if [ -z "${XDG_SPECIAL_DIRS+x}" ]; then
+    save_initial_values=true
+  fi
+
+  for d in $XDG_DIRS; do
+    var="XDG_${d}_DIR"
+    value="${!var}"
```

Comment:
This looks like a clever and elegant replacement for the horrible nested echo and eval magic. Thanks!

Reply:
Sure thing, I busted my noggin just trying to figure out what was happening with that other bit! :rofl: